index.bs

```diff
@@ -8,46 +8,190 @@ Repository: w3c/webrtc-dscp-exp
-control over the DSCP markings without affecting local prioritization.
+This feature was originally part of the [[!WEBRTC]] specification, but was
+removed in November 2019 due to lack of implementation
+experience. It is now part of this document.
+
+In addition, this specification adds fields to {{RTCRtpEncodingParameters}}
+that allow control over the DSCP markings without affecting local
+prioritization, and vice versa.
+
+# Priority and QoS Model # {#model}
+
+Many applications have multiple media flows of the same data type and
+often some of the flows are more important than others. WebRTC uses the
+priority and Quality of Service (QoS) framework described in
+[[!RTCWEB-TRANSPORT]] and [[!TSVWG-RTCWEB-QOS]] to provide priority and
+DSCP marking for packets that will help provide QoS in some networking
+environments. The priority setting can be used to indicate the relative
+priority of various flows. The priority API allows the JavaScript
+applications to tell the browser whether a particular media flow is high,
+medium, low or of very low importance to the application by setting the
+<code>priority</code> property of
+{{RTCRtpEncodingParameters}} objects to one of the
+values defined below.
+
+# Extensions for media priority # {#media-priority}
+
+## RTCPriorityType Enum ## {#rtc-priority-type}
+<pre class="idl">
+enum RTCPriorityType {
+  "very-low",
+  "low",
+  "medium",
+  "high"
+};
+</pre>
+<table data-link-for="RTCPriorityType" data-dfn-for="RTCPriorityType"
+class="simple">
+  <tbody>
+    <tr>
+      <th colspan="2">Enumeration description</th>
+    </tr>
+    <tr>
+      <td><dfn enum-value><code>very-low</code></dfn></td>
+      <td>See [[!RTCWEB-TRANSPORT]], Sections 4.1 and 4.2. Corresponds to "below
+      normal" as defined in [[!RTCWEB-DATA]].</td>
+    </tr>
+    <tr>
+      <td><dfn enum-value><code>low</code></dfn></td>
+      <td>See [[!RTCWEB-TRANSPORT]], Sections 4.1 and 4.2. Corresponds to
+      "normal" as defined in [[!RTCWEB-DATA]].</td>
+    </tr>
+    <tr>
+      <td><dfn enum-value><code>medium</code></dfn></td>
+      <td>See [[!RTCWEB-TRANSPORT]], Sections 4.1 and 4.2. Corresponds to "high"
+      as defined in [[!RTCWEB-DATA]].</td>
+    </tr>
+    <tr>
+      <td><dfn enum-value><code>high</code></dfn></td>
+      <td>See [[!RTCWEB-TRANSPORT]], Sections 4.1 and 4.2. Corresponds to "extra
+      high" as defined in [[!RTCWEB-DATA]].</td>
+    </tr>
+  </tbody>
+</table>
 
-IDL specification {#idl}
-====
+Applications that use this API should be aware that often better
+overall user experience is obtained by lowering the priority of things
+that are not as important rather than raising the priority of the things
+that are.
 
+## Extension to RTCRtpEncodingParameters ## {#encoding-parameters}
 <pre class='idl'>
 partial dictionary RTCRtpEncodingParameters {
-    RTCPriorityType networkPriority;  // Note: No default
+  RTCPriorityType priority;
+  RTCPriorityType localPriority;
```

Comment:
This is more than a move, adding a new `localPriority` field that did not exists in webrtc-pc.

Having three knobs to control two things seems like one too many (one of the three seem redundant).

E.g. I could already accomplish all combinations using `priority` and `networkPriority` alone.

Reply:
Yes. For symmetry, I would prefer to have localPriority and networkPriority only, but the 3-way thing preserves the text that had consensus with the IETF a long time ago (one control that influences both things).

The way to accomplish setting localPriority only without the value (setting priority to the desired value and also setting networkPrioirty to the default value) seems unintuitive and convoluted.

Suggestions?

Reply:
I'd pick 1 or 2 I think. I don't understand when I would want to set different priorities.